pkg/controllers/organization/organization_controller.go

```diff
@@ -50,10 +59,21 @@ func (r *OrganizationReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	orgStatus := initOrganizationStatus(org)
+	defer func() {
+		if statusErr := r.setStatus(ctx, org, orgStatus); statusErr != nil {
+			log.FromContext(ctx).Error(statusErr, "failed to set status")
+		}
+	}()
+
 	if err := r.reconcileNamespace(ctx, org); err != nil {
 		return ctrl.Result{}, err
 	}
 
+	orgStatus.SetConditions(greenhousesapv1alpha1.TrueCondition(greenhousesapv1alpha1.ReadyCondition, "", ""))
```

Comment:
This will always set the `Ready` condition to true. I think we should calculate this condition based on the status of the SCIM API availability. If no SCIM Config is set we should keep the `SCIMAPIAvailableCondition` as `Unknown` and the Ready State should be true, since the config is optional.
In case the config is provided then the `SCIMAPIAvailableCondition` should reflect if access works and then the result can be used to calculate the `Ready`